File: selenium_junit5_simple_with_dl_driver/src/test/java/org/o2f/pages/automationpractice/LandingPage.java
package org.o2f.pages.automationpractice;

import org.o2f.pages.BasePage;
import org.o2f.pages.automationpractice.components.HeaderComponent;
import org.openqa.selenium.WebDriver;

public class LandingPage extends BasePage {

    private HeaderComponent headerComponent;

    public LandingPage(WebDriver driver){
        super(driver);
        this.headerComponent = new HeaderComponent(driver);
    }

    public void isLoaded(){
        headerComponent.loginBtnShouldBeDisplayed();
        headerComponent.headerDiscountShouldBeDisplayed();
        //... in a real case we would have all the components pieces of all the components here
    }

    public void doSearchOf(String text){
        headerComponent.search(text);
    }

    public void navigateToAutomationPracticeHomePage(String envUrl){
        driver.get(envUrl);
    }
}

File: python-selenium-with-po-module/pages/automation_practice/landing_page.py
from selenium import webdriver

from config import settings
from pages.automation_practice.components.header_component import HeaderComponent
from pages.base_page import BasePage


class LandingPage(BasePage):

    def __init__(self, driver):
        super().__init__(driver)
        self.header_component = HeaderComponent(driver)

    def navigate_to_automation_practice(self):
        self.driver.get(settings.base_url)

    def perform_search(self, text: str):
        self.header_component.search(text)